inox2d/src/node/drawables.rs

```diff
@@ -0,0 +1,66 @@
+use glam::Mat4;
+
+use crate::node::{
+	components::{Composite, Drawable, TexturedMesh, TransformStore},
+	InoxNodeUuid,
+};
+use crate::puppet::World;
+
+/// Possible component combinations of a renderable node.
+///
+/// Future spec extensions go here.
+/// For user-defined custom nodes that can be rendered, as long as a subset of their components matches one of these variants,
+/// they will be picked up and enter the regular rendering pipeline.
+pub(crate) enum DrawableKind<'comps> {
+	TexturedMesh(TexturedMeshComponents<'comps>),
+	Composite(CompositeComponents<'comps>),
+}
+
+/// Pack of components for a TexturedMesh. "Part" in Inochi2D terms.
+pub struct TexturedMeshComponents<'comps> {
+	// Only the absolute part of `TransformStore` that the renderer backend may need.
+	pub transform: &'comps Mat4,
+	pub drawable: &'comps Drawable,
+	pub data: &'comps TexturedMesh,
+}
+
+/// Pack of components for a Composite node.
+pub struct CompositeComponents<'comps> {
+	// Only the absolute part of `TransformStore` that the renderer backend may need.
+	pub transform: &'comps Mat4,
+	pub drawable: &'comps Drawable,
+	pub data: &'comps Composite,
+}
+
+impl<'comps> DrawableKind<'comps> {
+	/// Tries to construct a renderable node data pack from the World of components:
+	/// - `None` if node not renderable.
+	/// - Panicks if component combinations invalid.
+	pub(crate) fn new(id: InoxNodeUuid, comps: &'comps World) -> Option<Self> {
```

Comment:
I'm not sure if this is the best place for this function or if it's even the right approach. It doesn't feel right to have to look at the components of a node only for it to return an enum that we have to then match against instead of doing it more directly. I don't have a nicer solution right now though but I'm probably gonna think about refactoring this part in the future.

Reply:
Related to [above](https://github.com/Inochi2D/inox2d/pull/86#discussion_r1698768218). This "component combination to enum variants" mapping is for following the Inochi2D spec. There should be a clear way of detecting "official" nodes, and then executing them in the "official" way.

Instead of removing this, what should be done further is a way to inject a end-user-provided routine to handle custom nodes, and before we hand over a node to a custom routine, we still need a way to extract the non-custom part and do what inox2d is supposed to do, which is what we have here, this `new()`.